mpd/internal/server/server.go

```diff
@@ -461,6 +485,29 @@ func (s *server) writeResponse(p *textproto.Conn, args []string, okLine string)
 			return
 		}
 		s.currentPlaylist.Delete(i)
+	case "deleteid":
+		if len(args) != 2 {
+			ack("wrong number of arguments")
+			return
+		}
+		id, err := strconv.Atoi(args[1])
+		if err != nil {
+			ack("invalid song ID")
+			return
+		}
+		s.idleEventc <- "playlist"
+		deleted := false
+		for i, song := range s.currentPlaylist.songs {
+			if song.id == id {
+				deleted = true
+				s.currentPlaylist.Delete(i)
+				break
+			}
+		}
+		if !deleted {
+			ackWithCode(50, "No such song")
```

Comment:
maybe define a const for `ACK_ERROR_NO_EXIST = 50` somewhere? It's ok to duplicate the consts from the mpd package, but we only need this one right now.